stl/inc/vector

```diff
@@ -2880,6 +2880,205 @@ namespace pmr {
     using vector = _STD vector<_Ty, polymorphic_allocator<_Ty>>;
 } // namespace pmr
 #endif // _HAS_CXX17
+
+#if _HAS_IF_CONSTEXPR
+// VARIABLE TEMPLATE _Is_vb_iterator
+template <class _Alloc>
+_INLINE_VAR constexpr bool _Is_vb_iterator<_Vb_const_iterator<_Alloc>> = true;
+
+template <class _Alloc>
+_INLINE_VAR constexpr bool _Is_vb_iterator<_Vb_iterator<_Alloc>> = true;
+
+template <class _InIt, class _OutIt>
+_CONSTEXPR20 _OutIt _Copy_vbool(_InIt _First, _InIt _Last, _OutIt _Dest) {
+    // copy [_First, _Last) to [_Dest, ...)
+    _Adl_verify_range(_First, _Last);
+
+    // Slow path as _First and _Dest are not aligned
+    if (_First._Myoff != _Dest._Myoff) {
+        for (; _First != _Last; ++_Dest, (void) ++_First) {
+            *_Dest = *_First;
+        }
+
+        return _Dest;
+    }
+
+    _Vbase* _UFirst = const_cast<_Vbase*>(_First._Myptr);
+    _Vbase* _UDest  = const_cast<_Vbase*>(_Dest._Myptr);
+    _Vbase* _ULast  = const_cast<_Vbase*>(_Last._Myptr);
+
+    // Fast path for less than _VBITS
+    if (_UFirst == _ULast) {
+        const auto _SourceMask = (_Vbase(-1) << _First._Myoff) & (_Vbase(-1) >> (_VBITS - _Last._Myoff));

```

Comment:
:facepalm: Did i just completely switch leftshift and right shift. and why did the new tests pass with that...